src/services/repo.cpp

```diff
@@ -75,6 +75,19 @@ std::vector<ELRepo> ELRepoFile::parse()
 {
     this->read();
 
+    return this->parseData();
+}
+
+std::vector<ELRepo> ELRepoFile::parse(const std::stringstream& ss)
+{
+    m_file = Glib::KeyFile::create();
+    m_file->load_from_data(ss.str().c_str());
+    return this->parseData();
+}
```

Comment:
Cool, now

* Add `m_repositories` member as a vector of ELRepo with a getter
* This function can return void and be private
* The getter should load the data into the member if the member is empty and return at the end

Reply:
Inside of `ELRepo?`

Reply:
Inside ELRepoFile